### PR TITLE
Refactor/842 rename layer list components

### PR DESCRIPTION
**Pull request by danielsteinkogler**
## Basics

- [x] I added a line to [changelog.md](/doc/changelog.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages and changelog.
- [x] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added automated tests or a [manual test protocol](../doc/tests/manual/protocol.md)
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)
- [ ] Exceptions to any guidelines are documented

## Review

- [x] I've tested the code
- [x] I've read through the whole code
- [x] I've read through the whole documentation
- [x] I've checked conformity to guidelines

### CODE REVIEW

File: frontend/src/features/map_planning/components/toolbar/LayerList.tsx [hunk 1]
```diff
@@ -1,109 +1,56 @@
...
-import IconButton from '@/components/Button/IconButton';
-import { NamedSlider } from '@/components/Slider/NamedSlider';
-import { ReactComponent as CaretDownIcon } from '@/icons/caret-down.svg';
-import { ReactComponent as CaretRightIcon } from '@/icons/caret-right.svg';
-import { ReactComponent as EyeOffIcon } from '@/icons/eye-off.svg';
-import { ReactComponent as EyeIcon } from '@/icons/eye.svg';
-import { useState } from 'react';
+import { LayerListItem } from './LayerListItem';
+import { LayerDto } from '@/bindings/definitions';
+// import IconButton from '@/components/Button/IconButton';
```
Comment (badnames) [inline frontend/src/features/map_planning/components/toolbar/LayerList.tsx:4]: Consider adding a short message saying why these imports are disabled, like you did in the component below.

File: frontend/src/features/map_planning/components/toolbar/LayerListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+interface LayerListProps {
+  /** layer which is controlled by this list element */
+  layer: LayerDto;
+  /** function that gets triggered when the layer is selected */
+  setSelectedLayer?: (layer: LayerDto) => void;
+  /** function that gets triggered when slider value is changed */
+  setLayerOpacity?: (name: LayerType, value: number) => void;
+  /** function that gets triggered when an alternative is selected */
+  setLayerAlternative?: (name: LayerType, value: string) => void;
+  /** list of names of the possible alternatives for this layer
```
Comment (badnames) [inline frontend/src/features/map_planning/components/toolbar/LayerListItem.tsx:21]:
```suggestion
  /** 
   * list of names of the possible alternatives for this layer
```

File: frontend/src/features/map_planning/components/toolbar/LayerListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import { ReactComponent as CaretRightIcon } from '@/icons/caret-right.svg';
+import { ReactComponent as EyeOffIcon } from '@/icons/eye-off.svg';
+import { ReactComponent as EyeIcon } from '@/icons/eye.svg';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+interface LayerListProps {
+  /** layer which is controlled by this list element */
+  layer: LayerDto;
+  /** function that gets triggered when the layer is selected */
```
Comment (badnames) [inline frontend/src/features/map_planning/components/toolbar/LayerListItem.tsx:15]:
```suggestion
  /** callback that gets triggered when the layer is selected */
```

File: frontend/src/features/map_planning/components/toolbar/LayerListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  setSelectedLayer?: (layer: LayerDto) => void;
+  /** function that gets triggered when slider value is changed */
```
Comment (badnames) [inline frontend/src/features/map_planning/components/toolbar/LayerListItem.tsx:17]:
```suggestion
  /** callback that gets triggered when slider value is changed */
```

File: frontend/src/features/map_planning/components/toolbar/LayerListItem.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  setLayerOpacity?: (name: LayerType, value: number) => void;
+  /** function that gets triggered when an alternative is selected */
```
Comment (badnames) [inline frontend/src/features/map_planning/components/toolbar/LayerListItem.tsx:19]:
```suggestion
  /** callback that gets triggered when an alternative is selected */
```

**Review by badnames — APPROVED** [review]
Good work. 
I could only find a few minor nitpicks.

**Comment by markus2330** [comment]
Thank you for refactoring. @badnames thx for the review! :heart:

**Pull request merged**